Update the OSPO Definition to Reflect Current Trends and Practices (#9)
todogroup/ospodefinition.org: README.md

*Nearly all organizations depend on technology created by the open source community*

## OSPO Definition

**[WHAT]** An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure. It is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; supporting an organization's efficiency in developing software through encouraging sustainable usage of existing open source components and, where appropriate, contributing enhancements back to these project; ensuring engineering effectiveness; ensuring legal compliance; and promoting and building community engagement.

> **[WHY]** An OSPO serves as a vital bridge between an organization and the open source community, helping to ensure that the organization is a good steward of open source software and can reap the benefits of open source adoption while minimizing risks.

Reviewer: I wonder if we could get more views on this - how common is it for organisations with <2-3 years of OSPO existence to also manage innersource? For us, we had an ISPO - and when we setup an OSPO, the ISPO transitioned over to us, because it made sense for the two to be together.
Innersource as a journey to open source / start of the path.

> **[WHY]** An OSPO serves as a vital bridge between an organization and the open source community, helping to ensure that the organization is a good steward of open source software and can reap the benefits of open source adoption while minimizing risks.

Author: Hey @cornelius @Sealjay thanks for sharing your thoughts! with regards to mentioning IneerSource, I agree that the term is used as a broader topic beyond organizations engaging in open source (e.g there are organizations doing innersource that does not necessarily do open source nor have "open source" teams) but I do know that the people behind OSPOs and similar open source initiatives are responsible for managing "cultural change within teh organization" that entails applying principles that are taken from innersoruce practices. Whether the OSPO engages with an ISPO (InnserSource Program Office) or not, they usually need to address the challenge of the organization's culture.

According to [2022 OSPO survey](https://github.com/todogroup/osposurvey/tree/main/2022), InnerSource is a topic that most of the respondents with an OSPO address.

![Survey](https://github.com/todogroup/ospodefinition.org/assets/43671777/ec85c446-df23-4239-a155-11fa566a8a29)

> **[WHY]** An OSPO serves as a vital bridge between an organization and the open source community, helping to ensure that the organization is a good steward of open source software and can reap the benefits of open source adoption while minimizing risks.

Reviewer 2: Should we put something about this in the **[What]** section?  We can acknowledge (or give a nod) to the fact that some OSPOs also steward and pay attention to InnerSource culture.  We don't have to go into detail but can then link from there to the InnerSource Commons.

> **[WHY]** An OSPO serves as a vital bridge between an organization and the open source community, helping to ensure that the organization is a good steward of open source software and can reap the benefits of open source adoption while minimizing risks.

Reviewer 3: Putting something in the What section would make sense to me.

> **[WHY]** An OSPO serves as a vital bridge between an organization and the open source community, helping to ensure that the organization is a good steward of open source software and can reap the benefits of open source adoption while minimizing risks.

Reviewer 2: Put [a suggestion](https://github.com/todogroup/ospodefinition.org/pull/9/files#r1222202734) out there.

**[HOW]** The way the people behind an OSPO achieve this is by creating and maintaining a framework covering the following aspects: strategy, governance, compliance, and community engagement. The OSPO's strategy focuses on aligning the organization's open source goals with its overall organization objectives.

> OSPOs can be formed in various sectors, regions, and organizational sizes including private and public organizations such as academics, NGOs, foundations, governments, and public administrations, as well as small, medium, and large companies. It's also important that there are many different names for an OSPO-like structure within an organization. The naming and concrete organizational form will strongly depend on the the practices of each organization. Organizing it as a "Program" or an "Office" is only one way to implement a structure which serves the purpose of an OSPO.

## Basic Segmentation

> There is no broad template for building an open source program that applies across all industries, or even across all companies in a single industry. However we can categorize the functions of an OSPO into different categories, depending on the nature of the organization and its motivators.

Author: ```suggestion
There is no broad template for building an open source program that applies across all industries, or even across all companies in a single industry. However we can categorize the functions of an OSPO into different categories, depending on the nature of the organization and its motivators. 

Please note that these categories do not necessarily need to be seen as silos. For instance, `Gov-` and `NonGov-Oriented` categories listed below could also be mixed with `educational-oriended` or have certain `business-roriented` motivations. The same happens with `business-oriented`category, that can provide training including educational-oriented goals.
```

Author: Addition based on the suggestion commented below: https://github.com/todogroup/ospodefinition.org/pull/9/files#r1210705025

Author: @cornelius what do you think of adding this extra line?

Reviewer 3: Looks good to me.

* `📈 Business-Oriented`: OSPOs established within for-profit organizations, such as enterprises, 
that are primarily focused on creating business value through the use of open source software. 

    * **Drivers of Motivation:** Innovation / Risk Management / Legal Compliance / Talent Retention

* `🎓 Educational-Oriented`: OSPOs established by educational institutions, such as universities 
or schools, that are focused on using open source software to support teaching, research and learning 
activities.

    * **Drivers of Motivation:** Curriculum development / Student Engagement / Open Data / Knowledge Sharing

> * `👩‍🏫 Business-Educational`: OSPOs established within for-profit organizations that provide training and certification and are likely to collaborate

Reviewer 3: Maybe instead of adding this mixed form of category, it would be better to add a sentence at the top which says that there can be mixed forms. Business-Eductional is not the only possible mix. Gov- and NonGov-Oriented could for example also be mixed with Educational or have certain business-related motivations.

with `educational-oriented` types.

>     * **Drivers of Motivation: hybrid between `📈 Business-Oriented`and `🎓 Educational-Oriented`

Reviewer 4: ```suggestion
    * **Drivers of Motivation:** hybrid between `📈 Business-Oriented`and `🎓 Educational-Oriented`
```

Author: Thanks for reviewing the punctuation! 🙏

* `🏛 Social-Gov-Oriented`: OSPOs established within a government or public administrators that are 
focused on using open source software to achieve social or public policy objectives (e.g serving 
citizens).

    * **Drivers of Motivation:** Interoperability / Open Data / Accessibility / Inclusion / Privacy / Security / Transparency

* `🌳 Social-NonGov-Oriented`: OSPOs established within non-governmental organizations (NGOs) or foundations that are dedicated 
to using open source software to create positive social change. 

    * **Drivers of Motivation:** Innovation / Interoperability / Social Justice / Disaster Relief / Humanitarian Aid / Environmental Protection / Sustainable Development

## Drivers of motivation

In this context, motivation is the driving force that fuels the entitiy to achieve its goals and 
reach its full potential. Based on the categories defined above, we can list some of the functions of an OSPO based on these motivators.

### Business-Oriented Segment

> Legal Risk Mitigation, Improving Engineers’ Practices and Enabling Financial Benefits. (A mnemonic: Fear, Love and Money)

Reviewer 5: @anajsana, Considering the mnemonic **Fear, Love and Money**... We tend to mitigate not "only legal risks".

What do you think about adding something like Open Source Risks mitigation (Legal, Security and Operational risks)

And just below a very brief description of how we already have Legal Risks for the other two "types of risks"

Author: We can just drop the ´legal´ from ´risk mitigation´ so it gives a broader view of the different risks that are, including security, operational, etc 🙂

> Legal Risk Mitigation, Improving Engineers’ Practices and Enabling Financial Benefits. (A mnemonic: Fear, Love and Money)

Author: ```suggestion
Risk Mitigation, Improving Engineers’ Practices and Enabling Financial Benefits. (A mnemonic: Fear, Love and Money)
```

# Open Source Program Office (OSPO) Definition and Guide

Open source is [intertwined in the development of modern technologies](https://www.synopsys.com/software-integrity/resources/analyst-reports/open-source-security-risk-analysis.html?intcmp=sig-blog-ossra22) and serves as the foundation of the vast majority of codebases across all industry sectors and technology areas, such as Artificial Inteligence and Machine Learning or Cybersecurity. 

*Nearly all organizations depend on technology created by the open source community*

## OSPO Definition

> 

Author: ```suggestion
> The pronunciation of *OSPO* would typically be said together as [oh -spoh]
```

> **[WHAT]** An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure. It is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, contributing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; supporting an organization's efficiency in developing software through encouraging sustainable usage of existing open source components and, where appropriate, contributing enhancements back to these project; when needed, guiding teams with open sourcing their software; ensuring engineering effectiveness; ensuring legal compliance; and promoting and building community engagement.

Author: ```suggestion

**[WHAT]** An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure. It is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, contributing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; supporting an organization's efficiency in developing software through encouraging sustainable usage of existing open source components and, where appropriate, contributing enhancements back to these project; when needed, guiding teams with open sourcing their software; ensuring engineering effectiveness; ensuring legal compliance; and promoting and building community engagement.
```

> **[WHAT]** An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure. It is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, contributing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; supporting an organization's efficiency in developing software through encouraging sustainable usage of existing open source components and, where appropriate, contributing enhancements back to these project; ensuring engineering effectiveness; ensuring legal compliance; and promoting and building community engagement.

Reviewer 6: Can we also mention OSPO's role on guiding teams with process and guidelines for "open sourcing a repository"? For example,

```suggestion
**[WHAT]** An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure. It is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, contributing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; supporting an organization's efficiency in developing software through encouraging sustainable usage of existing open source components and, where appropriate, contributing enhancements back to these project; when needed, guiding teams with open sourcing their software; ensuring engineering effectiveness; ensuring legal compliance; and promoting and building community engagement.
```

* Open Source Counselor: OSPOs can provide guidance and advice on the latest open source trends, licensing issues, and how to engage with open source projects, foundations, and communities. This can help organizations stay up-to-date with the rapidly changing open source landscape and ensure they are making informed decisions.

* Open Source Advocate: OSPOs can promote the use and/or contribution of open source and best practices across different organizational units. This can help organizations realize the benefits of open source as well as engaging people to contribute to open source projects or start new ones.

* Open Source Environmentalist: OSPOs can help organizations support and sustain open source projects in the long term by addressing issues such as security, maintenance, and project health. This can involve establishing policies and procedures for code review, security vulnerability management, and ongoing maintenance and support through funding and/or contributions. By doing so, OSPOs can help ensure that open source projects remain healthy and continue to benefit the wider community.

> 

Reviewer 6: Just a thought: Do we need to add a role for reviewing and approving compliant usage of open source software? In the above roles, I see Open Source Counselor provides advice on licensing issues. It would be good to specify who actually implements and ensures the license compliance. It can be a separate role or handled by one of the above specialist.

Author: Let's see if someone else has any thoughts on this idea! I'll also circle this back to the members from the TODO SC, who can also provide their views on this 👍

Reviewer 6: Did you get any suggestions on this? I feel we can add "ensuring and approving license compliance of open source softwares" can be added to one of the roles mentioned above, if not having a separate role for it.

# Open Source Program Office (OSPO) Definition and Guide

> Open source is [intertwined in the development of modern technologies](https://www.synopsys.com/software-integrity/resources/analyst-reports/open-source-security-risk-analysis.html?intcmp=sig-blog-ossra22) and serves as the foundation of the vast majority of codebases across all industry sectors and technology areas, such as Artificial Inteligence and Machine Learning or Cybersecurity. 

Author: changed the link to the OFE study 👍

*Nearly all organizations depend on technology created by the open source community*

## OSPO Definition
> The pronunciation of *OSPO* would typically be said together as [oh -spoh]

> **[WHAT]** An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure. It is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, contributing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; supporting an organization's efficiency in developing software through encouraging sustainable usage of existing open source components and, where appropriate, contributing enhancements back to these project; when needed, guiding teams with open sourcing their software; ensuring engineering effectiveness; ensuring legal compliance; and promoting and building community engagement.

Reviewer 2: ```suggestion
**[WHAT]** An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure. It is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, contributing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; supporting an organization's efficiency in developing software through encouraging sustainable usage of existing open source components and, where appropriate, contributing enhancements back to these project; when needed, guiding teams with open sourcing their software; ensuring engineering effectiveness; ensuring legal compliance; and promoting and building community engagement.
[InnerSource](https://innersourcecommons.org/) is a close sibling of open source and often collaborates with or is part of the OSPO.
```

**[HOW]** The way the people behind an OSPO achieve this is by creating and maintaining a framework covering the following aspects: strategy, governance, compliance, and community engagement. The OSPO's strategy focuses on aligning the organization's open source goals with its overall organization objectives.

> OSPOs can be formed in various sectors, regions, and organizational sizes including private and public organizations such as academics, NGOs, foundations, governments, and public administrations, as well as small, medium, and large companies. It's also important that there are many different names for an OSPO-like structure within an organization. The naming and concrete organizational form will strongly depend on the the practices of each organization. Organizing it as a "Program" or an "Office" is only one way to implement a structure which serves the purpose of an OSPO.

> ## Basic Segmentation

Author: The main reason for this is that categorizations can help to better understand and reference the different OSPOs created by looking at their motivators. For instance, there are certain topics relevant to corporate OSPOs that are completely out of the scope of other OSPOs created within universities or governments, and vice versa. Having a set of basic categories for people to use can help to differentiate and find similarities in OSPOs being established across sectors.

To give all of you context, this comes from a past issue discussion and work on categories started in OSPOlogy repo: https://github.com/todogroup/ospology/issues/261 and was also implemented in the ospo-book project in the taxonomy section, reviewed by contributors of the project: https://github.com/todogroup/ospology/blob/main/ospo-book/chapters/taxonomy.md

> ## Basic Segmentation

Author: Following up on this @lhawthorn 🙂  Since the review period is already past due and this comment is the only one missing, I'll add in this section a disclaimer mentioning that the *basic segmentation* is in *beta version*. This way, we can focus on your comment and work on this section to decide whether or not can be improved or should be deleted in the future. It can also be a good chance to bring more people from other OSPO communities with deep experience in other angles (gov, NGOs) to provide input.

# Open Source Program Office (OSPO) Definition and Guide

> Open source is [intertwined in the development of modern technologies](https://www.synopsys.com/software-integrity/resources/analyst-reports/open-source-security-risk-analysis.html?intcmp=sig-blog-ossra22) and serves as the foundation of the vast majority of codebases across all industry sectors and technology areas, such as Artificial Inteligence and Machine Learning or Cybersecurity. 

Author: ```suggestion
Open source is [intertwined in the development of modern technologies](https://openforumeurope.org/open-source-impact-study/) and serves as the foundation of the vast majority of codebases across all industry sectors and technology areas, such as Artificial Inteligence and Machine Learning or Cybersecurity. 
```

*Nearly all organizations depend on technology created by the open source community*

## OSPO Definition

> > The pronunciation of *OSPO* would typically be said together as [oh -spoh]

Author: ```suggestion
> The pronunciation of *OSPO* would typically be said together as [ah -spoh]
```

[InnerSource](https://innersourcecommons.org/) is a close sibling of open source and often collaborates with or is part of the OSPO.

**[WHY]** An OSPO serves as a vital bridge between an organization and the open source community, helping to ensure that the organization is a good steward of open source software and can reap the benefits of open source adoption while minimizing risks.

**[WHO]** OSPOs are composed of people (open source specialists) wearing different hats:

> * Open Source Enabler: OSPOs can help organizations navigate the cultural, process, and tool changes required to engage with the open source community effectively. This can involve educating teams/ units, establishing new processes and workflows, and adopting new tools and technologies.

Author: ```suggestion
* Open Source Enabler: OSPOs can help organizations navigate the cultural, process, and tool changes required to engage with the open source community effectively. This can involve educating teams/units, establishing new processes and workflows, and adopting new tools and technologies.
```

> * Open Source Counselor: OSPOs can provide guidance and advice on the latest open source trends, licensing issues, and how to engage with open source projects, foundations, and communities. This can help organizations stay up-to-date with the rapidly changing open source landscape and ensure they are making informed decisions.

Author: ```suggestion
* Open Source Counselor: OSPOs can provide guidance and advice on the latest open source trends, licensing trends, and how to engage with open source projects, foundations, and communities. This can help organizations stay up-to-date with the rapidly changing open source landscape and ensure they are making informed decisions.
```

* Open Source Advocate: OSPOs can promote the use and/or contribution of open source and best practices across different organizational units. This can help organizations realize the benefits of open source as well as engaging people to contribute to open source projects or start new ones.

* Open Source Environmentalist: OSPOs can help organizations support and sustain open source projects in the long term by addressing issues such as security, maintenance, and project health. This can involve establishing policies and procedures for code review, security vulnerability management, and ongoing maintenance and support through funding and/or contributions. By doing so, OSPOs can help ensure that open source projects remain healthy and continue to benefit the wider community.

> **[HOW]** The way the people behind an OSPO achieve this is by creating and maintaining a framework covering the following aspects: strategy, governance, compliance, and community engagement. The OSPO's strategy focuses on aligning the organization's open source goals with its overall organization objectives.

Author: ```suggestion
**[HOW]** The way the people behind an OSPO achieve this is by creating and maintaining a framework covering the following aspects: strategy, governance, compliance, and community engagement. The OSPO's strategy focuses on aligning the organization's open source goals with its overall organization objectives and works with all lines of organization units/groups
```

> OSPOs can be formed in various sectors, regions, and organizational sizes including private and public organizations such as academics, NGOs, foundations, governments, and public administrations, as well as small, medium, and large companies. It's also important that there are many different names for an OSPO-like structure within an organization. The naming and concrete organizational form will strongly depend on the the practices of each organization. Organizing it as a "Program" or an "Office" is only one way to implement a structure which serves the purpose of an OSPO.

> ## Basic Segmentation

Author: ```suggestion
## Basic Segmentation (Beta Version)
```

## OSPO Definition

> An open source program office (OSPO) serves as the center of competency for an organization's open source operations and structure, and is responsible for defining and implementing strategies and policies to guide these efforts. This can include setting policies around code use, distribution, selection, auditing, and other key areas; providing education and training to people (inside and outside the organization) involved in open source activities; ensuring legal compliance; and promoting and building community engagement.

Reviewer 7: suggestion to be inserted after the clause "ensuring legal compliance;"

supporting an organization's efficiency in developing software through encouraging sustainable consumption of and reuse of existing open source components and, where appropriate, contributing enhancements back to these projects; ....

* Open Source Advocate: OSPOs can promote the use and/or contribution of open source and best practices across different organizational units. This can help organizations realize the benefits of open source as well as engaging people to contribute to open source projects or start new ones.

* Open Source Environmentalist: OSPOs can help organizations support and sustain open source projects in the long term by addressing issues such as security, maintenance, and project health. This can involve establishing policies and procedures for code review, security vulnerability management, and ongoing maintenance and support through funding and/or contributions. By doing so, OSPOs can help ensure that open source projects remain healthy and continue to benefit the wider community.

**[HOW]** The way the people behind an OSPO achieve this is by creating and maintaining a framework covering the following aspects: strategy, governance, compliance, and community engagement. The OSPO's strategy focuses on aligning the organization's open source goals with its overall organization objectives.

> **[WHY]** An OSPO serves as a vital bridge between an organization and the open source community, helping to ensure that the organization is a good steward of open source software and can reap the benefits of open source adoption while minimizing risks.

Reviewer 8: Set **[WHY]** section before **[HOW]** or **[WHO]** ?  
According to the Golden Circle model, this will enable us to better comprehend the complete definition in a more seamless manner.

> OSPOs can be formed in various sectors, regions, and organizational sizes including private and public organizations such as academics, NGOs, foundations, governments, and public administrations, as well as small, medium, and large companies. It's also important to note that the term "Program" in OSPOs can sometimes be excluded (Open Source Office).

> ## Basic Segmentation

Reviewer 9: My understanding of this section and how the text is phrased is a bit confusing to me. I believe this section intends to categorize different types of OSPOs in different host organizations. So, as we know every OSPO is different and even more so across segments. So, I suggest to remove words like "typical" and "usual" as they seem to contradict this intention.

Reviewer 7: Proposed in SC mtg: add a "stub" for other segments that we may know about and call for contributors from other communities to share their wisdom in these sections

*Each open source program office is custom-configured based on its particular business, products, and goals.*

***
## OSPO Landscape

> Organizations that have an OSPO or hires open source especialists to manage open source operations and structure: https://landscape.todogroup.org/

Author: Good question @lhawthorn! The OSPO Landscape is reviewed frequently (new PRs and issues) by TODO Maintainers + dedicated staff within CNCF that helps with all LF landscape infrastructure and new developments. 

Everyone can submit a PR to include their organization's logo to the Lanscape if they are OSPO adopters and or TODO GM, or can ping the TODO PM to submit the PR  instead. However, we can't include logos that we don't have the right permission to include in the landscape, even though we know that the organization has an OSPO. It must be a representative of that organization who contacts us to request their logo in the landscape, that way, we can add it 👍 

Re. cards: That is how it should look at this point, but you can request a new feature by submitting an issue, and will pass it to the development team to take into account 🙂

* [Comcast](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/comcast.md)
* [Meta](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/facebook.md)
* [Microsoft](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/microsoft.md)
* [Postman](https://dev.to/postman/how-postmans-building-their-open-source-program-office-ospo-hgf)
* [Red Hat](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/redhat.md)
* [RIT](https://todogroup.org/guides/casestudies/rit/)
* [Salesforce](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/salesforce.md)
* [SAP](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/sap.md)
* [UC Santa Cruz](https://research.redhat.com/blog/article/building-a-university-ospo-bolstering-academic-research-through-open-source/)
* [Uber](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/uber.md)

> * [Yahoo](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/oath.md)

Reviewer 4: Seems like the link is for Oath Inc. Yahoo use case is not included in `https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies`

```suggestion
* [Oath Inc](https://github.com/todogroup/todogroup.github.io/blob/master/content/en/guides/casestudies/oath.md)
```

Author: Did a bit of research! Looks like Oath Inc is the umbrella name of Yahoo! & AOL (initially owned by Verizon). But that has changed since 2021.

From [Wikipedia page](https://en.wikipedia.org/wiki/Yahoo!_Inc._(2017%E2%80%93present)):

> On May 3, 2021, Verizon announced that the Verizon Media would be acquired by [Apollo Global Management](https://en.wikipedia.org/wiki/Apollo_Global_Management) for roughly $5 billion, and would simply be known as Yahoo following the closure of the deal, with Verizon retaining a minor 10% stake in the new group

Author: so, maybe we should keep it as yahoo 🤔 what do you think @UlisesGascon ?

Reviewer 4: Yes! makes sense @anajsana. Thanks for the research 🙂
